### ISSUE TITLE
pip package should unpack to codalab-cli and contain the worker code

### ISSUE THREAD

**Issue by percyliang** [codalab/codalab-worksheets#314]
1) Currently, it's called `codalab` rather than `codalab-cli` (the directory in site-packages).
2) I'd like to be able to run `cl-worker` (add as a top-level script like `cl`)

**Comment by percyliang** [comment]
Remember to update the wiki for 2).

**Comment by maxwang7** [comment]
@percyliang 

Proposed solutions to 1 (related SO post http://stackoverflow.com/a/36012825):

- Change the directory name.
- Move `codalab/` into `codalab/codalab-cli`.
- Symlink `codalab-cli/` to `codalab/`

Though the third option seems rather hacky, it's also the least disruptive. I feel like the first two options would require extensively testing of our CLI scripts as well as if there are any assumptions by codalab-worksheets and codalab-deployment make of the directory structure.

There is no way to map the name of a package to a name other than its directory name: http://stackoverflow.com/a/36012825.

**Comment by maxwang7** [comment]
Rename worker to codalab-worker, have codalab-cli depend on codalab-worker, have two separate setup.py files. Leave codalab as codalab, just move it into PyPI repo named codalab (don't push new versions to codalab-cli anymore).

**Comment by maxwang7** [comment]
[ ] Draw requirements in setup.py from requirements.txt

**Comment by maxwang7** [comment]
https://www.reddit.com/r/Python/comments/3kdy10/why_does_pep_8_discourage_underscores_in_package/

**Comment by maxwang7** [comment]
I've run through CLI commands and run the python tests again, what are other things I could do to test the path changes? I was trying to run the worker command but I've had a problem in my instance prior to these changes, so it's not a perfect test. Still, the worker is checking in with the REST API ok.

**Comment by maxwang7** [comment]
Ran into a problem with separating worker and codalab packages, would like feedback on proposed solution: `setuptools` requires that setup files are named `setup.py`, so I can't create a setup file for the worker package in the project-level directory (http://stackoverflow.com/questions/8898352/multiple-projects-using-multiple-setup-py-scripts). Creating a `setup.py` file in the `codalab-cli/worker` directory also doesn't work, because `setuptools` can only make packages out of the subdirectories of the directory that contains the `setup.py` file.  The proposed solution is to move the contents currently inside the worker package under another directory, `worker`. The resulting directory structure would look like:

setup.py (for codalab package)
worker/
    setup.py (for worker package)
    worker/
        ... (original contents of codalab-cli/worker)

I don't think this change should break anything, since I've changed the import paths to reference the worker's package name and so scripts shouldn't need to depend on the specific file structure of `codalab-cli`. Is that evaluation correct? Are there other things I should consider?
